api/src/main/java/com/cloud/network/Network.java

```diff
@@ -412,12 +412,16 @@ public void setIp6Address(String ip6Address) {
 
     String getGateway();
 
+    void setGateway(String gateway);
```

Comment:
@shwstppr 
In this PR, users are able to create ROUTED network with cidrsize (gateway and cidr are null), ACS will allocate a IPv4 subnet (like ipv6 subnet) to the network. In this case, the gateway/cidr needs to be updated accordingly.

th methods `setCidr` and `setGateway` already exist in `NetworkVO`. However, the class passed to `ExternalGuestNetworkGuru` class is `Network` , not `NetworkVO`, therefore I promoted it to `Network` interface, otherwise we need to change the type `Network` to "`? extends Network`" (or other) in all related classes.